spatialdata/_core/transform.py

```diff
@@ -1,86 +1,526 @@
+        raise NotImplementedError()
 
-class Transform:
+    # @property
+    # def ndim(self) -> Optional[int]:
+    #     return self._ndim
+
+
+class Translation(BaseTransformation):
     def __init__(
         self,
-        translation: Optional[ArrayLike] = None,
-        scale_factors: Optional[ArrayLike] = None,
+        translation: Optional[Union[ArrayLike, List[Any]]] = None,
         ndim: Optional[int] = None,
-    ):
+    ) -> None:
         """
-        class for storing transformations, required to align the different spatial data layers.
+        class for storing translation transformations.
         """
         if translation is None:
-            if scale_factors is not None:
-                translation = np.zeros(len(scale_factors), dtype=float)
-            elif ndim is not None:
-                translation = np.zeros(ndim, dtype=float)
-            else:
-                raise ValueError("Either translation or scale_factors or ndim must be specified")
+            assert ndim is not None
+            translation = np.ones(ndim, dtype=float)
+
+        if ndim is None:
+            assert translation is not None
+            if isinstance(translation, list):
+                translation = np.array(translation)
+            ndim = len(translation)
+
+        assert type(translation) == np.ndarray
+        assert len(translation) == ndim
+        self.translation = translation
+        self._ndim = ndim
+
+    @property
+    def ndim(self) -> Optional[int]:
+        return self._ndim
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {
+            "type": "translation",
+            "translation": self.translation.tolist(),
+        }
+
+    def transform_points(self, points: ArrayLike) -> ArrayLike:
+        return points + self.translation
+
+    def inverse(self) -> BaseTransformation:
+        return Translation(translation=-self.translation)
+
+    def to_affine(self) -> Affine:
+        assert self.ndim == 2
+        m: ArrayLike = np.vstack(
+            (np.hstack((np.array([[1.0, 0.0], [0.0, 1.0]]), self.translation.reshape(2, 1))), np.array([0.0, 0.0, 1.0]))
+        )
+        return Affine(affine=m)
+
+
+class Scale(BaseTransformation):
+    def __init__(
+        self,
+        scale: Optional[Union[ArrayLike, List[Any]]] = None,
+        ndim: Optional[int] = None,
+    ) -> None:
+        """
+        class for storing scale transformations.
+        """
+        if scale is None:
+            assert ndim is not None
+            scale = np.ones(ndim, dtype=float)
+
+        if ndim is None:
+            assert scale is not None
+            if isinstance(scale, list):
+                scale = np.array(scale)
+            ndim = len(scale)
+
+        assert type(scale) == np.ndarray
+        assert len(scale) == ndim
+        self.scale = scale
+        self._ndim = ndim
+
+    @property
+    def ndim(self) -> Optional[int]:
+        return self._ndim
+
+    def to_dict(self) -> Dict[str, Any]:
```

Comment:
general comment borderline nitpick, but I would explicilty type dictionaries, not with `Any`. This one especially should just be string or sequences of floats

Reply:
Agree, I will do it.